configure.py

```diff
@@ -1429,6 +1433,120 @@ def bakefile_test_sources(file, sources):
 
     f.close()
 
+def gen_cmake(build_config, using_mods, cc, options):
+    def create_target_rules(sources):
+        target = {'sources': {}, 'frameworks': set(), 'libs': set()}
+        for source in sources:
+            target['sources'][source] = {'isa_flags': set()}
+        return target
+
+    def add_target_details(target, using_mod):
+        libs_or_frameworks_needed = False
+        for source_path in target['sources']:
+            for mod_source in using_mod.source:
+                if source_path == mod_source :
+                    libs_or_frameworks_needed = True
+                    for isa in using_mod.need_isa:
+                        target['sources'][source_path]['isa_flags'].add(cc.isa_flags[isa])
+        if libs_or_frameworks_needed == True:
+            if options.os in using_mod.libs:
+                for lib in using_mod.libs[options.os]:
+                    target['libs'].add(lib)
+            if options.os in using_mod.frameworks:
+                for framework in using_mod.frameworks[options.os]:
+                    target['frameworks'].add('"-framework '+framework+'"')
+
+    def generate_target_sources_list(fd, targetName, target):
+        fd.write('set(' + targetName + '\n')
+        for source in target['sources']:
+            fd.write('    ${CMAKE_CURRENT_LIST_DIR}' + os.sep + os.path.normpath(source) + '\n')
+        fd.write(')\n\n')
+
+    def generate_target_source_files_isa_properties(fd, target):
+        for source in target['sources']:
+            joinedIsaFlags = ' '.join(target['sources'][source]['isa_flags'])
```

Comment:
We use snake_case [for variable names in Python](https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables)